### PR TITLE
[Lexer] Don't include backtick length into comment length

### PR DESCRIPTION
**Pull request by rintaro**
Although "backtick" is a kind of trivia piece, `Token::getCommentRange` doesn't take it into account. Invalid `Token::getCommentRange` used to cause compiler crash.

rdar://problem/42492793
https://bugs.swift.org/browse/SR-8315

### CODE REVIEW

**Comment by rintaro** [comment]
CC @omochi

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: lib/Parse/Lexer.cpp [hunk 1]
```diff
@@ -283,6 +283,8 @@ void Lexer::formToken(tok Kind, const char *TokStart, bool MultilineString) {
       return Piece.isComment();
     });
     for (auto End = LeadingTrivia.end(); Iter != End; Iter++) {
+      if (Iter->getKind() == TriviaKind::Backtick)
+        break;
```
Comment (nkcsgexi) [inline lib/Parse/Lexer.cpp]: Can we whitelist comment kinds instead of blacklisting backsticks? That way makes more sense to me.

Comment (rintaro) [reply lib/Parse/Lexer.cpp]: We have to whitelist all kind of comments and whitespaces.

Comment (nkcsgexi) [reply lib/Parse/Lexer.cpp]: ok, can we add some comments here why backtick is different? It's kind of missing connection that here `CommentLength` means more than comments in syntax tree.

Comment (rintaro) [reply lib/Parse/Lexer.cpp]: Added!

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: lib/Parse/Lexer.cpp (hunk 1 above)
Comment (nkcsgexi) [reply lib/Parse/Lexer.cpp]: Thanks!

**Review by nkcsgexi — APPROVED** [review]

**Pull request merged**